src/cache_inode/cache_inode_readdir.c

```diff
@@ -223,6 +225,7 @@ cache_inode_operate_cached_dirent(cache_entry_t *directory,
 			/* try to rename--no longer in-place */
 			dirent3 = gsh_malloc(sizeof(cache_inode_dir_entry_t)
 					     + newnamesize);
+      number_of_dirent_entries++;
```

Comment:
Check this. We are doing rename. Will old entry be deleted ? If so, does incrementing make sense ?
Also check if you would have to decrement the value in CACHE_INODE_DIRENT_OP_REMOVE.

Reply:
In CACHE_INODE_DIRENT_OP_REMOVE case we are not really removing an entry. I decrementing the count exactly, where we are removing the entry. In the case of rename also I am incrementing the counter, as entry is allocated.